src/librustc_session/lint/builtin.rs

```diff
@@ -474,15 +474,34 @@ declare_lint! {
 
 declare_lint! {
     pub INDIRECT_STRUCTURAL_MATCH,
-    // defaulting to allow until rust-lang/rust#62614 is fixed.
-    Allow,
-    "pattern with const indirectly referencing non-structural-match type",
+    Warn,
+    "constant used in pattern contains value of non-structural-match type",
```

Comment:
The lint name and description do not align. Is this specific to indirections or not?

Reply:
the indirect refers to the fact that the constant's type is structural match, but the value contains a non-structural-match value somewhere in a field or sth

Reply:
I'd say this means the type isn't structural match, we just accidentally declared it as such.

Reply:
The problem is that right now structural match is a shallow feature. We can change this, but that's a different discussion to be had imo.

Reply:
Yeah that's a discussion for https://github.com/rust-lang/rust/issues/74446.
I just don't see how it can make any sense as a shallow feature.^^

Reply:
But the comment here is about it being confusing that the lint name contains the term "indirect" but the lint description does not. I think that can be fixed with a better description or a better lint name.